### PR TITLE
Introduce KiteLogger class

### LINKED ISSUE #46: Replace enableLogging util with injectable Logger class
instead of relying on a function (`enableLogging`) that depends on a kite being an `EventEmitter` instance, create a new `KiteLogger` class and allow `logger` option to be passed while creating a kite:

```js
const { Logger, Kite } = require('kite.js')
const { DebugLevel } = require('kite.js/lib/constants')
const Kite = new Kite({
  name: 'foo',
  logger: new Logger({
    name: 'foo',
    level: DebugLevel.INFO,
    handlers: {
      error: (...args) => sendToPaperTrail('error', ...args),
      warn: (...args) => sendToPaperTrail('warn', ...args),
      info: (...args) => sendToPaperTrail('info', ...args),
    }
  })
})
```
- a logger will use `console.error`, `console.warn`, `console.info` as its handlers if any of them is not being passed

  ```js
  const logger = new Logger({name: 'foo'})
  logger.critical('message is here, hello world')
  // => [foo] CRITICAL      message is here, hello world
  ```

- a logger will expose **6** methods in which the passed `handlers` are being used:

  - `logger.critical(...args): void` -> handlers.error
  - `logger.error(...args): void` -> handlers.error
  - `logger.warning(...args): void` -> handlers.warn
  - `logger.notice(...args): void` -> handlers.info
  - `logger.info(...args): void` -> handlers.info
  - `logger.debug(...args): void` -> handlers.info

- a logger will accept a `level` option while it's being created which will be used to decide when one of the methods are gonna produce logs (each method has its own `DebugLevel` associated with it)

  - These are the defined `DebugLevel`s available:

    ```js
    export const DebugLevel = {
      CRITICAL: 0,
      ERROR: 1,
      WARNING: 2,
      NOTICE: 3,
      INFO: 4,
      DEBUG: 5,
    }
    ```
  - This will allow each exposed method to work since all debug levels are lower than `DebugLevel.DEBUG` which is `0`:

    ```js
    const logger = new Logger({
      name: 'foo',
      level: DebugLevel.DEBUG
    })

    logger.debug('Hello world - debug') // DebugLevel.DEBUG == 5
    logger.warn('Hello world - warn') // DebugLevel.WARNING == 2
    logger.critical('Hello world - critical') // DebugLevel.CRITICAL == 0
    // => [kite] DEBUG      Hello world - debug
    // => [kite] WARNING    Hello world - warning
    // => [kite] CRITICAL   Hello world - critical
    ```
  - But this will only work for critical as only the `DebugLevel.CRITICAL` is lower or equal to 0.
    ```js
    const logger = new Logger({
      name: 'foo',
      level: DebugLevel.CRITICAL
    })

    logger.debug('Hello world - debug') // DebugLevel.DEBUG == 5
    logger.warn('Hello world - warn') // DebugLevel.WARNING == 2
    logger.critical('Hello world - critical') // DebugLevel.CRITICAL == 0
    // => [kite] CRITICAL   Hello world - critical
    ```

### PR DESCRIPTION
**Pull request by usirin**
Fixes #46

### CODE REVIEW

File: src/kite/base.js [hunk 1]
```diff
@@ -71,11 +74,12 @@ class BaseKite extends Emitter {
     this.messageScrubber = new MessageScrubber({ kite: this })
 
     this.proto.on(Event.request, req => {
-      this.ready(() => this.ws.send(JSON.stringify(req)))
-      this.emit(Event.debug, 'Sending: ', JSON.stringify(req))
+      const message = JSON.stringify(req)
```
Comment (gokmen) [inline src/kite/base.js]: 👍

File: src/kite/base.js [hunk 2]
```diff
@@ -204,6 +208,7 @@ class BaseKite extends Emitter {
 
   onError(err) {
     this.emit(Event.error, 'Websocket error!')
```
Comment (gokmen) [inline src/kite/base.js]: should we keep this? this was also for logger afaik.

File: src/server/sockjs/index.js [hunk 3]
```diff
@@ -44,8 +53,3 @@ class Server extends Emitter {
     }
   }
 }
-
-Server.scheme = 'http'
-Server.secureScheme = 'https'
```
Comment (gokmen) [inline src/server/sockjs/index.js]: 👍

File: src/kite/base.js (hunk 2 above)
Comment (usirin) [inline src/kite/base.js]: `error` event is being used in Kontrol [here](https://github.com/koding/kite.js/blob/master/src/kontrol/base.js#L45)

**Pull request merged**